x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_panel.tsx

```diff
@@ -170,14 +179,21 @@ export function SuggestionPanel({
     ? stagedPreview.visualization.activeId
     : activeVisualizationId;
 
-  const { suggestions, currentStateExpression } = useMemo(() => {
+  const { suggestions, currentStateExpression, currentStateError } = useMemo(() => {
     const newSuggestions = getSuggestions({
       datasourceMap,
       datasourceStates: currentDatasourceStates,
       visualizationMap,
       activeVisualizationId: currentVisualizationId,
       visualizationState: currentVisualizationState,
     })
+      .filter(({ visualizationId, visualizationState: suggestionVisualizationState }) => {
+        const suggestionVisualization = visualizationMap[visualizationId];
+        // filter out visualizations with errors
+        return (
+          suggestionVisualization.getErrorMessages(suggestionVisualizationState, frame) == null
```

Comment:
This is not checking whether the datasource contains errors

Reply:
The check is below the new suggestion creation.
It may make sense to exit earlier one if that fails before producing new suggestions: I'll push an update.